In [2]:
import pandas as pd

In [74]:
pwd

'/Users/iankim/Library/Mobile Documents/com~apple~CloudDocs/data/airbnb/hotel_level_mathcing_feb11'

In [1]:
ls

Method_1and2.ipynb
hotel_population_20211102_493.dta
korea_hotel_govnt.csv
한국관광협회중앙회_호텔업등급결정현황_20220208.xlsx


## 1.전처리

<pre>

1. master: hotel_population_20211102_493.dta
 --> 전화 번호 및 호텔 이름 공백 제거
 
2. target: korea_hotel_govnt.csv
 --> 전화 번호 및 호텔 이름 공백 제거
 
</pre>

In [18]:
master = pd.read_stata("hotel_population_20211102_493.dta")
master_simple = master[['id_regist','phone','hotelname', 'sigungu']]

temp_phone = [a.replace(" ", "") for a in master_simple['phone'].tolist()]
master_simple['phone'] = temp_phone

temp_name = [a.replace(" ", "") for a in master_simple['hotelname'].tolist()]
master_simple['hotelname'] = temp_name

master_simple.head(5)

/Users/iankim/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """
/Users/iankim/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


id_regist       phone  hotelname sigungu
0      26945  0336107000    썬크루즈리조트     강릉시
1      27029  0336559900  포씨즌비치관광호텔     강릉시
2      27768  0335341004     대주관광호텔     동해시
3      27787  0335327373  망상해오름가족호텔     동해시
4      28360  0335707899   삼척온천관광호텔     삼척시

In [26]:
target = pd.read_csv("korea_hotel_govnt.csv")
target_simple = target[['결정 등급', '호텔명', '전화번호', '주소']]

temp_phone2 = [a.replace("-", "") for a in target_simple['전화번호'].tolist()]
target_simple['전화번호'] = temp_phone2

temp_name2 = [a.replace(" ", "") for a in target_simple['호텔명'].tolist()]
target_simple['호텔명'] = temp_name2

target_simple = target_simple.rename({'전화번호':'phone'}, axis = 'columns')
target_simple = target_simple.rename({'호텔명':'hotelname'}, axis = 'columns')

target_simple.head(5)

/Users/iankim/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """
/Users/iankim/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


결정 등급     hotelname       phone                         주소
0    2성  하슬라아트월드뮤지엄호텔  0336449411      강원도 강릉시 강동면 율곡로 1441 
1    4성       쏠비치호텔삼척  0338037955    강원도 삼척시 수로부인길 453 (갈천동)
2    3성       뉴동해관광호텔  0335339215       강원도 동해시 평릉길 1  (천곡동)
3    4성       롯데리조트속초  0336341101     강원도 속초시 대포항길 186 (대포동)
4    3성      동해현진관광호텔  0335392000  강원도 동해시 동굴로 125-3 . (천곡동)

## 2. Method (1) 전화번호

In [27]:
test = pd.merge(master_simple, target_simple, on = 'phone')

In [28]:
test

id_regist       phone         hotelname_x sigungu 결정 등급  \
0       24290  0313390064              뉴스카이호텔     용인시    2성   
1       24322  0316938412             소설스미스호텔     용인시    1성   
2       49824  0552239800            사과나무관광호텔     창원시    1성   
3        6937  0512600000  페어필드바이메리어트부산송도비치호텔      서구    3성   
4       10237  0519829700             호텔센트럴베이     수영구    3성   
5        1837   024476900              호텔스타건대     광진구    1성   
6        3607   023137003              호텔밀하우스    서대문구    1성   
7         463  0263537700           오라카이대학로호텔     종로구    4성   
8       18936  0522099000             굿모닝관광호텔      동구    3성   
9       13817  0324290204             호텔베이204     남동구    2성   
10      13733  0327262000        오크우드프리미어인천호텔     연수구    5성   
11      13678  0327170700         호텔스카이파크인천송도     연수구    3성   
12      12905  0327528333              관광호텔레이      중구    1성   
13      12821  0327778780             다이아몬드호텔      중구    1성   
14      12941  0327648993             시크릿관광호텔      중구    2성   
15      12841  0327528080               호텔ORA      중구    3성   
16      38648  0619241500      벨메르바이한화호텔앤드리조트     여수시    4성   
17      38572  0616803000              히든베이호텔     여수시    4성   
18      35409  0632446400             호텔오페라21     전주시    1성   
19      51388  0432132332               세종스파텔     청주시    1성   

           hotelname_y                               주소  
0               뉴스카이호텔          경기도 용인시 처인구 백암면 고안로 48   
1              소설스미스호텔       경기도 용인시 기흥구 민속촌로 57  (보라동)  
2               사과나무호텔  경상남도 창원시 마산합포구 가포신항북로 16  (가포동)  
3   페어필드바이메리어트부산송도비치호텔        부산광역시 서구 송도해변로 113  (암남동)  
4              호텔센트럴베이       부산광역시 수영구 광안해변로 189  (광안동)  
5               호텔스타건대       서울특별시 광진구 광나루로30길 3  (구의동)  
6               호텔밀하우스       서울특별시 서대문구 연세로2나길 5  (창천동)  
7            오라카이대학로호텔         서울특별시 종로구 율곡로 180  (원남동)  
8                호텔굿모닝                울산광역시 동구 바드래1길 15  
9              호텔베이204   인천광역시 남동구 인하로507번길 21-1  (구월동)  
10        오크우드프리미어인천호텔       인천광역시 연수구 컨벤시아대로 165 (송도동)  
11         호텔스카이파크인천송도      인천광역시 연수구 컨벤시아대로 233  (송도동)  
12              레이관광호텔         인천광역시 중구 을왕로58번길 7 (을왕동)  
13             다이아몬드호텔        인천광역시 중구 월미로 229  (북성동1가)  
14               호텔시크릿             인천광역시 중구 월미로233번길 11  
15                호텔오라         인천광역시 중구 공항서로 345  (남북동)  
16      벨메르바이한화호텔앤드리조트         전라남도 여수시 웅천남4로 17  (웅천동)  
17              히든베이호텔       전라남도 여수시 신월로 496-25  (신월동)  
18             오페라21호텔   전라북도 전주시 덕진구 정언신로 186  (우아동2가)  
19               세종스파텔     충청북도 청주시 청원구 내수읍 신기초정로 699

## 3. Method (2) 호텔명

In [29]:
test2 = pd.merge(master_simple, target_simple, on = 'hotelname')
test2

id_regist       phone_x           hotelname sigungu 결정 등급     phone_y  \
0       19710    0312332996               자니비호텔     수원시    1성  0312218008   
1       19655  050714478881                해시호텔     수원시    1성  0312474301   
2       24290    0313390064              뉴스카이호텔     용인시    2성  0313390064   
3       24322    0316938412             소설스미스호텔     용인시    1성  0316938412   
4       22550    0316117462              월드관광호텔     평택시    1성  0313537833   
5       49345    0315324497               호텔갤러리     포천시    1성  0315448008   
6       46473    0552929902               노마드호텔     김해시    2성  0553267700   
7       50043                              리안호텔     창원시    2성  0557135577   
8       50079    0517443003               여우비호텔     창원시    2성  0552439999   
9       45896    0556441211               브룩스호텔     통영시    1성  0556441212   
10      43136    0544769994              킴스관광호텔     구미시    2성   023790520   
11      12035    0536027111            호텔인터불고대구     수성구    5성  0536027114   
12      10721                        하운드호텔기장일광점     기장군    2성  0517233322   
13       6937    0512600000  페어필드바이메리어트부산송도비치호텔      서구    3성  0512600000   
14      10237    0519829700             호텔센트럴베이     수영구    3성  0519829700   
15       6711    0514669101               코모도호텔      중구    4성  0514619101   
16       6094    0234528663              제이비스호텔     강남구    2성  0318364500   
17       1837     024476900              호텔스타건대     광진구    1성   024476900   
18       3550    0222300367            신라스테이서대문    서대문구    3성  0222300700   
19       3607     023137003              호텔밀하우스    서대문구    1성   023137003   
20        456                              글루호텔     종로구    2성  0220248400   
21        463    0263537700           오라카이대학로호텔     종로구    4성  0263537700   
22      13817    0324290204             호텔베이204     남동구    2성   029557400   
23      13817    0324290204             호텔베이204     남동구    2성  0324290204   
24      13733    0327262000        오크우드프리미어인천호텔     연수구    5성  0327262000   
25      13678    0327170700         호텔스카이파크인천송도     연수구    3성  0327170700   
26      12821    0327778780             다이아몬드호텔      중구    1성  0327778780   
27      38648    0619241500      벨메르바이한화호텔앤드리조트     여수시    4성  0619241500   
28      38572    0616803000              히든베이호텔     여수시    4성  0616803000   
29      35943       4677000             호텔은파팰리스     군산시    3성  0634677000   
30      33536                          에버그린관광호텔     논산시    1성  0318557900   
31      51416    0439086677              백제관광호텔     청주시    1성  0418350870   
32      51388    0432132332               세종스파텔     청주시    1성  0432132332   

                                주소  
0       경기도 수원시 권선구 장다리로 75  (권선동)  
1     경기도 수원시 팔달구 갓매산로 72-1  (고등동)  
2          경기도 용인시 처인구 백암면 고안로 48   
3       경기도 용인시 기흥구 민속촌로 57  (보라동)  
4       경기도 화성시 향남읍 발안공단로5길 7-23    
5        경기도 포천시 소흘읍 죽엽산로 432-17    
6        경상남도 김해시 인제로91번길 5  (어방동)  
7      경상남도 창원시 진해구 명제로 478  (제덕동)  
8   경상남도 창원시 마산합포구 월영동11길 22 (해운동)  
9        경상남도 통영시 통영해안로 407  (정량동)  
10      서울특별시 종로구 평창문화로 136  (평창동)  
11        대구광역시 수성구 팔현길 212  (만촌동)  
12          부산광역시 기장군 일광면 해송3로 2    
13       부산광역시 서구 송도해변로 113  (암남동)  
14      부산광역시 수영구 광안해변로 189  (광안동)  
15         부산광역시 중구 중구로 151  (영주동)  
16     경기도 의정부시 청사로6번길 19-7  (신곡동)  
17      서울특별시 광진구 광나루로30길 3  (구의동)  
18        서울특별시 서대문구 충정로 76  (미근동)  
19      서울특별시 서대문구 연세로2나길 5  (창천동)  
20        서울특별시 종로구 율곡로 228  (이화동)  
21        서울특별시 종로구 율곡로 180  (원남동)  
22        서울특별시 도봉구 도봉로 710  (방학동)  
23  인천광역시 남동구 인하로507번길 21-1  (구월동)  
24      인천광역시 연수구 컨벤시아대로 165 (송도동)  
25     인천광역시 연수구 컨벤시아대로 233  (송도동)  
26       인천광역시 중구 월미로 229  (북성동1가)  
27        전라남도 여수시 웅천남4로 17  (웅천동)  
28      전라남도 여수시 신월로 496-25  (신월동)  
29        전라북도 군산시 은파순환길 20  (나운동)  
30        경기도 양주시 장흥면 권율로 190-45    
31            충청남도 부여군 부여읍 북포로 108  
32    충청북도 청주시 청원구 내수읍 신기초정로 699

<pre>
id_regist: 43136 킴스관광호텔 이름만 같음 --> 주소가 다름
id_regist: 6094 제이비스호텔 이름만 같음 --> 주소가 다름
id_regist: 33536 에버그린관광호텔 이름만 같음 --> 주소가 다름
id_regist: 51416 백제관광호텔 이름만 같음 --> 주소가 다름

id_regist: 13817 2성 --> 중복 삭제하고 남기기
</pre>

## 4. Concat

<pre>
1. test1 과 test2 concat
2. duplicates drop by id_regist
3. [43136, 6094, 33536, 51416] 호텔은 삭제

--> merged_by_1and2.csv 로 내보내기
</pre>

In [30]:
test_to_concat = test[['id_regist', 'hotelname_x', '결정 등급']]
test_to_concat = test_to_concat.rename({'hotelname_x':'hotelname'}, axis = 'columns')

In [32]:
test2_to_concat = test2[['id_regist', 'hotelname', '결정 등급']]

In [66]:
merged_by_1and2 = pd.concat([test_to_concat, test2_to_concat], ignore_index=True)

In [67]:
merged_by_1and2 = merged_by_1and2.drop_duplicates(subset=['id_regist'])

In [68]:
merged_by_1and2

id_regist           hotelname 결정 등급
0       24290              뉴스카이호텔    2성
1       24322             소설스미스호텔    1성
2       49824            사과나무관광호텔    1성
3        6937  페어필드바이메리어트부산송도비치호텔    3성
4       10237             호텔센트럴베이    3성
5        1837              호텔스타건대    1성
6        3607              호텔밀하우스    1성
7         463           오라카이대학로호텔    4성
8       18936             굿모닝관광호텔    3성
9       13817             호텔베이204    2성
10      13733        오크우드프리미어인천호텔    5성
11      13678         호텔스카이파크인천송도    3성
12      12905              관광호텔레이    1성
13      12821             다이아몬드호텔    1성
14      12941             시크릿관광호텔    2성
15      12841               호텔ORA    3성
16      38648      벨메르바이한화호텔앤드리조트    4성
17      38572              히든베이호텔    4성
18      35409             호텔오페라21    1성
19      51388               세종스파텔    1성
20      19710               자니비호텔    1성
21      19655                해시호텔    1성
24      22550              월드관광호텔    1성
25      49345               호텔갤러리    1성
26      46473               노마드호텔    2성
27      50043                리안호텔    2성
28      50079               여우비호텔    2성
29      45896               브룩스호텔    1성
30      43136              킴스관광호텔    2성
31      12035            호텔인터불고대구    5성
32      10721          하운드호텔기장일광점    2성
35       6711               코모도호텔    4성
36       6094              제이비스호텔    2성
38       3550            신라스테이서대문    3성
40        456                글루호텔    2성
49      35943             호텔은파팰리스    3성
50      33536            에버그린관광호텔    1성
51      51416              백제관광호텔    1성

In [69]:
merged_by_1and2.shape

(38, 3)

In [70]:
erase = [43136, 6094, 33536, 51416]
to_erase = merged_by_1and2.query('id_regist ==@ erase')
to_erase.index.tolist()

[30, 36, 50, 51]

In [71]:
merged_by_1and2 = merged_by_1and2.drop(to_erase.index)
merged_by_1and2.reset_index(drop = True, inplace = True)
merged_by_1and2

id_regist           hotelname 결정 등급
0       24290              뉴스카이호텔    2성
1       24322             소설스미스호텔    1성
2       49824            사과나무관광호텔    1성
3        6937  페어필드바이메리어트부산송도비치호텔    3성
4       10237             호텔센트럴베이    3성
5        1837              호텔스타건대    1성
6        3607              호텔밀하우스    1성
7         463           오라카이대학로호텔    4성
8       18936             굿모닝관광호텔    3성
9       13817             호텔베이204    2성
10      13733        오크우드프리미어인천호텔    5성
11      13678         호텔스카이파크인천송도    3성
12      12905              관광호텔레이    1성
13      12821             다이아몬드호텔    1성
14      12941             시크릿관광호텔    2성
15      12841               호텔ORA    3성
16      38648      벨메르바이한화호텔앤드리조트    4성
17      38572              히든베이호텔    4성
18      35409             호텔오페라21    1성
19      51388               세종스파텔    1성
20      19710               자니비호텔    1성
21      19655                해시호텔    1성
22      22550              월드관광호텔    1성
23      49345               호텔갤러리    1성
24      46473               노마드호텔    2성
25      50043                리안호텔    2성
26      50079               여우비호텔    2성
27      45896               브룩스호텔    1성
28      12035            호텔인터불고대구    5성
29      10721          하운드호텔기장일광점    2성
30       6711               코모도호텔    4성
31       3550            신라스테이서대문    3성
32        456                글루호텔    2성
33      35943             호텔은파팰리스    3성

In [72]:
merged_by_1and2.shape

(34, 3)

In [73]:
merged_by_1and2.to_csv("merged_by_1and2.csv")

---------

## 5. Method (3) 자료준비 

In [76]:
hotel_left458 = pd.read_stata("hotel_left458.dta")
hotel_left458.to_csv("hotel_left458.csv")

In [79]:
target.head(10)

NO     등급 인정처 호텔업 구분 결정 등급      등급 결정일   지역                   호텔명  객실수  \
0   24  한국관광협회중앙회  소형호텔업    2성  2021-12-13  강원도      하슬라 아트 월드 뮤지엄 호텔   23   
1   83  한국관광협회중앙회  관광호텔업    4성  2021-07-14  강원도             쏠비치 호텔 삼척  217   
2   89  한국관광협회중앙회  관광호텔업    3성  2021-07-07  강원도               뉴동해관광호텔   63   
3   90  한국관광협회중앙회  관광호텔업    4성  2021-07-07  강원도             롯데 리조트 속초  392   
4  101  한국관광협회중앙회  관광호텔업    3성  2021-06-17  강원도              동해현진관광호텔   82   
5  112  한국관광협회중앙회  관광호텔업    4성  2021-06-08  강원도                쏠비치 양양  214   
6  137  한국관광협회중앙회  관광호텔업    4성  2021-04-30  강원도  하이원 팰리스호텔 (구. 하이원호텔)  197   
7  189     한국관광공사  관광호텔업    3성  2020-10-15  강원도                  시티호텔   49   
8  235     한국관광공사  관광호텔업    3성  2020-04-21  강원도           베니키아 호텔 비즈인   65   
9  242     한국관광공사  관광호텔업    5성  2020-03-30  강원도    인터컨티넨탈 알펜시아 평창 리조트  238   

                          주소   Latitude   Longitude          전화번호  \
0      강원도 강릉시 강동면 율곡로 1441   37.705602  129.008001  033-644-9411   
1    강원도 삼척시 수로부인길 453 (갈천동)  37.471898  129.159248  033-803-7955   
2       강원도 동해시 평릉길 1  (천곡동)  37.529464  129.102202  033-533-9215   
3     강원도 속초시 대포항길 186 (대포동)  38.176011  128.607212  033-634-1101   
4  강원도 동해시 동굴로 125-3 . (천곡동)  37.518859  129.117718  033-539-2000   
5    강원도 양양군 손양면 선사유적로 678    38.087918  128.665748  033-670-3622   
6     강원도 정선군 고한읍 고한7길 399    37.173717  128.842806  033-590-7111   
7        강원도 원주시 건강로 4 (반곡동)  37.326721  127.986168  033-747-1000   
8            강원도 원주시 만대로 7-1  37.341131  127.923792  033-748-0100   
9       강원도 평창군 대관령면 솔봉로 325  37.661610  128.676336  033-339-1225   

                                                홈페이지  
0                          https://museumhaslla.com/  
1                  https://www.sonohotelsresorts.com  
2                                 http://hotelnd.com  
3                        http://www.lotteresort.com/  
4                                                NaN  
5           https://www.sonohotelsresorts.com/sb/yy/  
6                               http://www.high1.com  
7         http://www.wonjucitihotel.com/kor/index.do  
8                          http://www.biz-inn.co.kr/  
9  https://www.alpensia.com/accomm/ic/convenience.do

In [81]:
target_to_qgis = target[['호텔명', 'Latitude', 'Longitude', '결정 등급', 'NO']]

In [82]:
target_to_qgis.to_csv("target_to_qgis.csv")

## 6. Method 1,2 and 3 concat

In [84]:
m3 = pd.read_csv("intersection_100X100.csv")

In [93]:
m3 = m3[['id_regist', 'hotelname', '호텔등급']]
m3 = m3.rename({'호텔등급':'결정 등급'}, axis = 'columns')

In [94]:
m3

id_regist            hotelname 결정 등급
0        1001  (주)조선호텔앤리조트 포포인츠 명동    4성
1        1094           호텔 미드시티 명동    3성
2        6065              호텔 인 나인    4성
3        9203   (주)신세계조선호텔그랜드조선 부산    5성
4       12873            더프린스 관광호텔    2성
5       17810              리베라유성호텔    3성
6       20426    ㈜신세계조선호텔 그래비티서울판교    4성
7       22578           ㈜호성 평택관광호텔    2성
8       23098         소노캄 고양 웨스트타워    5성
9       37251       무주덕유산리조트(티롤호텔)    4성
10      40835            쏠비치 호텔 진도    4성
11      41213             송도코모도 호텔    2성

In [95]:
merged_by_upto3 = pd.concat([merged_by_1and2, m3], ignore_index=True)

In [96]:
merged_by_upto3

id_regist            hotelname 결정 등급
0       24290               뉴스카이호텔    2성
1       24322              소설스미스호텔    1성
2       49824             사과나무관광호텔    1성
3        6937   페어필드바이메리어트부산송도비치호텔    3성
4       10237              호텔센트럴베이    3성
5        1837               호텔스타건대    1성
6        3607               호텔밀하우스    1성
7         463            오라카이대학로호텔    4성
8       18936              굿모닝관광호텔    3성
9       13817              호텔베이204    2성
10      13733         오크우드프리미어인천호텔    5성
11      13678          호텔스카이파크인천송도    3성
12      12905               관광호텔레이    1성
13      12821              다이아몬드호텔    1성
14      12941              시크릿관광호텔    2성
15      12841                호텔ORA    3성
16      38648       벨메르바이한화호텔앤드리조트    4성
17      38572               히든베이호텔    4성
18      35409              호텔오페라21    1성
19      51388                세종스파텔    1성
20      19710                자니비호텔    1성
21      19655                 해시호텔    1성
22      22550               월드관광호텔    1성
23      49345                호텔갤러리    1성
24      46473                노마드호텔    2성
25      50043                 리안호텔    2성
26      50079                여우비호텔    2성
27      45896                브룩스호텔    1성
28      12035             호텔인터불고대구    5성
29      10721           하운드호텔기장일광점    2성
30       6711                코모도호텔    4성
31       3550             신라스테이서대문    3성
32        456                 글루호텔    2성
33      35943              호텔은파팰리스    3성
34       1001  (주)조선호텔앤리조트 포포인츠 명동    4성
35       1094           호텔 미드시티 명동    3성
36       6065              호텔 인 나인    4성
37       9203   (주)신세계조선호텔그랜드조선 부산    5성
38      12873            더프린스 관광호텔    2성
39      17810              리베라유성호텔    3성
40      20426    ㈜신세계조선호텔 그래비티서울판교    4성
41      22578           ㈜호성 평택관광호텔    2성
42      23098         소노캄 고양 웨스트타워    5성
43      37251       무주덕유산리조트(티롤호텔)    4성
44      40835            쏠비치 호텔 진도    4성
45      41213             송도코모도 호텔    2성

In [97]:
merged_by_upto3.to_csv("merged_by_upto3.csv")

## 6. Method (4) Naver Scraping 

In [98]:
import requests
from bs4 import BeautifulSoup

In [112]:
hotel_446 = pd.read_csv("hotel_left446.csv")

In [116]:
hotel_list = hotel_446['hotelname'].tolist()

In [118]:
url_list = ["https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=" + i for i in hotel_list]

In [119]:
url_list

['https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=목시 서울 인사동 호텔',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=나인트리 프리미어 호텔 인사동',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=호텔DM(호텔디엠)',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=신신호텔1',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=남산시티호텔',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=호텔 파레스',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=아스토리아호텔',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=뉴오리엔탈관광호텔',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=나인트리 호텔 동대문',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=에이원호텔',
 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=비젼관광호텔',
 'https://search.naver.com/search.nave

In [145]:
rate = []
for url in url_list:
    response = requests.get(url)
    html = response.text
    soup = BeautifulSoup(html, 'html.parser')
    div_hotel = soup.find('div', attrs={'class':"nx_hotel sc_new"})
    
    if div_hotel == None:
        rate.append("None")
    
    elif len(div_hotel) == 1:
        div_grade = div_hotel.find('span', attrs={'class':"grade"})
        if div_grade == None:
            rate.append("None")
        else:
            temp = div_grade.get_text()
            rate.append(temp)

    else:
        rate.append("Multiple")

In [146]:
len(url_list)

446

In [147]:
len(rate)

446

In [149]:
hotel_446['naver_rate'] = rate

In [151]:
hotel_446.to_csv("hotel_446_naver.csv")